### ISSUE TITLE
A NPE problem has occurred

### ISSUE THREAD

**Issue by Liuxboy** [GIScience/openrouteservice#531]
Hello,man, when I run this project from github with IDEA embed Tomcat, there is a NPE problem as follow, and in the line 38 of app.config: **"source": "openrouteservice/src/main/files/heidelberg.osm.gz"**, 
this source file has not been packaged into the osr war, Where should this configuration item be placed relative to the war package?? 

```
  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v2.0.4.RELEASE)

31 五月 15:09:38 INFO [ors.Application] - Starting Application v5.0.1 on 030994BG01 with PID 20440 (D:\Gitspace\openrouteservice\openrouteservice\target\ors\WEB-INF\classes started by 030994 in D:\Development\Tomcat8\bin)
31 五月 15:09:38 DEBUG [ors.Application] - Running with Spring Boot v2.0.4.RELEASE, Spring v5.0.8.RELEASE
31 五月 15:09:38 INFO [ors.Application] - No active profile set, falling back to default profiles: default
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
31 五月 15:09:49 INFO [ors.Application] - Started Application in 15.702 seconds (JVM running for 45.857)
<?xml version="1.0" ?>
<Configuration>
  <Properties>
    <Property name="filename">/var/log/ors/ors-logs.log</Property>
  </Properties>
</Configuration>
[2019-05-31 03:09:49,142] Artifact openrouteservice:war exploded: Artifact is deployed successfully
[2019-05-31 03:09:49,142] Artifact openrouteservice:war exploded: Deploy took 41,275 milliseconds
31 五月 15:09:51 INFO [routing.RoutingProfileManager] -  Total - 634 MB, Free - 222.26 MB, Max: 1.75 GB, Used - 411.74 MB
31 五月 15:09:51 INFO [routing.RoutingProfileManager] -       
31 五月 15:09:51 INFO [routing.RoutingProfileManager] - ====> Initializing profiles from 'openrouteservice/src/main/files/heidelberg.osm.gz' (2 threads) ...
31 五月 15:09:51 INFO [routing.RoutingProfileManager] -                               
31 五月 15:09:51 INFO [routing.RoutingProfileManager] -                
31 五月 15:09:51 INFO [routing.RoutingProfileManager] - [1] Profiles: 'driving-hgv', location: 'graphs\hgv'.
31 五月 15:09:51 INFO [routing.RoutingProfileManager] - [2] Profiles: 'driving-car', location: 'graphs\car'.
31 五月 15:10:23 ERROR [routing.RoutingProfileManager] - java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
31 五月 15:10:23 INFO [routing.RoutingProfileManager] - [3] Profiles: 'cycling-regular', location: 'graphs\bike-regular'.
java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at heigit.ors.routing.RoutingProfileManager.initialize(RoutingProfileManager.java:216)
	at heigit.ors.routing.RoutingProfileManager.getInstance(RoutingProfileManager.java:64)
	at heigit.ors.servlet.listeners.ORSInitContextListener.lambda$contextInitialized$1(ORSInitContextListener.java:43)
	at heigit.ors.servlet.listeners.ORSInitContextListener$$Lambda$295/1863649279.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at heigit.ors.routing.graphhopper.extensions.core.CoreLMAlgoFactoryDecorator.loadOrDoWork(CoreLMAlgoFactoryDecorator.java:278)
	at heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.loadOrPrepareCoreLM(ORSGraphHopper.java:623)
	at heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.postProcessing(ORSGraphHopper.java:539)
	at com.graphhopper.GraphHopper.process(GraphHopper.java:692)
	at com.graphhopper.GraphHopper.importOrLoad(GraphHopper.java:661)
	at heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.importOrLoad(ORSGraphHopper.java:156)
	at heigit.ors.routing.RoutingProfile.initGraphHopper(RoutingProfile.java:158)
	at heigit.ors.routing.RoutingProfile.<init>(RoutingProfile.java:103)
	at heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:37)
	at heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:21)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	... 1 more
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLMAlgoFactoryDecorator.loadOrDoWork(CoreLMAlgoFactoryDecorator.java:274)
	... 17 more
Caused by: java.lang.NullPointerException
	at heigit.ors.routing.graphhopper.extensions.edgefilters.core.AvoidBordersCoreEdgeFilter.restrictedCountry(AvoidBordersCoreEdgeFilter.java:64)
	at heigit.ors.routing.graphhopper.extensions.edgefilters.core.AvoidBordersCoreEdgeFilter.accept(AvoidBordersCoreEdgeFilter.java:53)
	at heigit.ors.routing.graphhopper.extensions.edgefilters.EdgeFilterSequence.accept(EdgeFilterSequence.java:28)
	at heigit.ors.routing.graphhopper.extensions.edgefilters.EdgeFilterSequence.accept(EdgeFilterSequence.java:28)
	at com.graphhopper.storage.BaseGraph$EdgeIterable.next(BaseGraph.java:979)
	at heigit.ors.routing.graphhopper.extensions.core.TarjansCoreSCCAlgorithm.strongConnect(TarjansCoreSCCAlgorithm.java:152)
	at heigit.ors.routing.graphhopper.extensions.core.TarjansCoreSCCAlgorithm.findComponents(TarjansCoreSCCAlgorithm.java:108)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage.createLandmarks(CoreLandmarkStorage.java:251)
	at heigit.ors.routing.graphhopper.extensions.core.PrepareCoreLandmarks.doWork(PrepareCoreLandmarks.java:134)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLMAlgoFactoryDecorator$1.run(CoreLMAlgoFactoryDecorator.java:264)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	... 8 more
31 五月 15:10:28 ERROR [routing.RoutingProfileManager] - java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
31 五月 15:10:28 INFO [routing.RoutingProfileManager] - [4] Profiles: 'cycling-mountain', location: 'graphs\bike-mountain'.
java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at heigit.ors.routing.RoutingProfileManager.initialize(RoutingProfileManager.java:216)
	at heigit.ors.routing.RoutingProfileManager.getInstance(RoutingProfileManager.java:64)
	at heigit.ors.servlet.listeners.ORSInitContextListener.lambda$contextInitialized$1(ORSInitContextListener.java:43)
	at heigit.ors.servlet.listeners.ORSInitContextListener$$Lambda$295/1863649279.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at heigit.ors.routing.graphhopper.extensions.core.CoreLMAlgoFactoryDecorator.loadOrDoWork(CoreLMAlgoFactoryDecorator.java:278)
	at heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.loadOrPrepareCoreLM(ORSGraphHopper.java:623)
	at heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.postProcessing(ORSGraphHopper.java:539)
	at com.graphhopper.GraphHopper.process(GraphHopper.java:692)
	at com.graphhopper.GraphHopper.importOrLoad(GraphHopper.java:661)
	at heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.importOrLoad(ORSGraphHopper.java:156)
	at heigit.ors.routing.RoutingProfile.initGraphHopper(RoutingProfile.java:158)
	at heigit.ors.routing.RoutingProfile.<init>(RoutingProfile.java:103)
	at heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:37)
	at heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:21)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	... 1 more
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLMAlgoFactoryDecorator.loadOrDoWork(CoreLMAlgoFactoryDecorator.java:274)
	... 17 more
Caused by: java.lang.NullPointerException
	at heigit.ors.routing.graphhopper.extensions.edgefilters.core.AvoidBordersCoreEdgeFilter.restrictedCountry(AvoidBordersCoreEdgeFilter.java:64)
	at heigit.ors.routing.graphhopper.extensions.edgefilters.core.AvoidBordersCoreEdgeFilter.accept(AvoidBordersCoreEdgeFilter.java:53)
	at heigit.ors.routing.graphhopper.extensions.edgefilters.EdgeFilterSequence.accept(EdgeFilterSequence.java:28)
	at heigit.ors.routing.graphhopper.extensions.edgefilters.EdgeFilterSequence.accept(EdgeFilterSequence.java:28)
	at com.graphhopper.storage.BaseGraph$EdgeIterable.next(BaseGraph.java:979)
	at heigit.ors.routing.graphhopper.extensions.core.TarjansCoreSCCAlgorithm.strongConnect(TarjansCoreSCCAlgorithm.java:152)
	at heigit.ors.routing.graphhopper.extensions.core.TarjansCoreSCCAlgorithm.findComponents(TarjansCoreSCCAlgorithm.java:108)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage.createLandmarks(CoreLandmarkStorage.java:251)
	at heigit.ors.routing.graphhopper.extensions.core.PrepareCoreLandmarks.doWork(PrepareCoreLandmarks.java:134)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLMAlgoFactoryDecorator$1.run(CoreLMAlgoFactoryDecorator.java:264)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	... 8 more
31 五月 15:10:33 INFO [routing.RoutingProfileManager] - [3] FlagEncoders: 1, bits used [UNKNOWN]/64.
31 五月 15:10:33 INFO [routing.RoutingProfileManager] - [3] Capacity: [UNKNOWN]. (edges - 39664, nodes - 28778)
31 五月 15:10:33 INFO [routing.RoutingProfileManager] - [3] Total time: 9.581s.
31 五月 15:10:33 INFO [routing.RoutingProfileManager] - [3] Finished at: 2019-05-31 15:10:33.
31 五月 15:10:33 INFO [routing.RoutingProfileManager] -                               
31 五月 15:10:33 INFO [routing.RoutingProfileManager] - [5] Profiles: 'cycling-road', location: 'graphs\bike-road'.
31 五月 15:10:33 INFO [bike.CommonBikeFlagEncoder] - NextGen RoadBike FlagEncoder is active...
31 五月 15:10:37 INFO [routing.RoutingProfileManager] - [4] FlagEncoders: 1, bits used [UNKNOWN]/64.
31 五月 15:10:37 INFO [routing.RoutingProfileManager] - [4] Capacity: [UNKNOWN]. (edges - 39704, nodes - 28803)
31 五月 15:10:37 INFO [routing.RoutingProfileManager] - [4] Total time: 9.438s.
31 五月 15:10:37 INFO [routing.RoutingProfileManager] - [4] Finished at: 2019-05-31 15:10:37.
31 五月 15:10:37 INFO [routing.RoutingProfileManager] -                               
31 五月 15:10:37 INFO [routing.RoutingProfileManager] - [6] Profiles: 'cycling-electric', location: 'graphs\bike-electric'.
31 五月 15:10:42 INFO [routing.RoutingProfileManager] - [5] FlagEncoders: 1, bits used [UNKNOWN]/64.
31 五月 15:10:42 INFO [routing.RoutingProfileManager] - [5] Capacity: [UNKNOWN]. (edges - 39319, nodes - 28576)
31 五月 15:10:42 INFO [routing.RoutingProfileManager] - [5] Total time: 8.66s.
31 五月 15:10:42 INFO [routing.RoutingProfileManager] - [5] Finished at: 2019-05-31 15:10:42.
31 五月 15:10:42 INFO [routing.RoutingProfileManager] -                               
31 五月 15:10:42 INFO [routing.RoutingProfileManager] - [7] Profiles: 'foot-walking', location: 'graphs\walking'.
31 五月 15:10:45 INFO [routing.RoutingProfileManager] - [6] FlagEncoders: 1, bits used [UNKNOWN]/64.
31 五月 15:10:45 INFO [routing.RoutingProfileManager] - [6] Capacity: [UNKNOWN]. (edges - 39664, nodes - 28778)
31 五月 15:10:45 INFO [routing.RoutingProfileManager] - [6] Total time: 8.143s.
31 五月 15:10:45 INFO [routing.RoutingProfileManager] - [6] Finished at: 2019-05-31 15:10:45.
31 五月 15:10:45 INFO [routing.RoutingProfileManager] -                               
31 五月 15:10:45 INFO [routing.RoutingProfileManager] - [8] Profiles: 'foot-hiking', location: 'graphs\hiking'.
31 五月 15:10:51 INFO [routing.RoutingProfileManager] - [7] FlagEncoders: 1, bits used [UNKNOWN]/64.
31 五月 15:10:51 INFO [routing.RoutingProfileManager] - [7] Capacity: [UNKNOWN]. (edges - 41449, nodes - 30359)
31 五月 15:10:51 INFO [routing.RoutingProfileManager] - [7] Total time: 8.944s.
31 五月 15:10:51 INFO [routing.RoutingProfileManager] - [7] Finished at: 2019-05-31 15:10:51.
31 五月 15:10:51 INFO [routing.RoutingProfileManager] -                               
31 五月 15:10:51 INFO [routing.RoutingProfileManager] - [9] Profiles: 'wheelchair', location: 'graphs\wheelchair'.
31 五月 15:10:54 INFO [routing.RoutingProfileManager] - [8] FlagEncoders: 1, bits used [UNKNOWN]/64.
31 五月 15:10:54 INFO [routing.RoutingProfileManager] - [8] Capacity: [UNKNOWN]. (edges - 41450, nodes - 30359)
31 五月 15:10:54 INFO [routing.RoutingProfileManager] - [8] Total time: 9.146s.
31 五月 15:10:54 INFO [routing.RoutingProfileManager] - [8] Finished at: 2019-05-31 15:10:54.
31 五月 15:10:54 INFO [routing.RoutingProfileManager] -                               
31 五月 15:11:03 INFO [routing.RoutingProfileManager] - [9] FlagEncoders: 1, bits used [UNKNOWN]/64.
31 五月 15:11:03 INFO [routing.RoutingProfileManager] - [9] Capacity: [UNKNOWN]. (edges - 25175, nodes - 14088)
31 五月 15:11:03 INFO [routing.RoutingProfileManager] - [9] Total time: 12.258s.
31 五月 15:11:03 INFO [routing.RoutingProfileManager] - [9] Finished at: 2019-05-31 15:11:03.
31 五月 15:11:03 INFO [routing.RoutingProfileManager] -                             
```

**Comment by rabidllama** [comment]
This seems to be the same as on #545

**Comment by rabidllama** [comment]
I have just put a comment on #545 that should tell you how to fix the null pointer problem (a line in the app.config needs updating).
For the location of the osm file - I don't believe this gets packaged in the war, and so the path is to the location on the system running it. When running locally I tend to use the full path to the file to make things a bit easier to follow

**Comment by rabidllama** [comment]
Fixed in Release 5.0.2
